Add an equivalent to iTerm's "show/hide all windows" hotkey (Fix for: #2205)

rolandpeelen opened the pull request:
First time contributor, so if I'm violating something, please let me know.

This feature should fix: https://github.com/ghostty-org/ghostty/issues/2205 (and is coincidentally basically the only thing holding me back from daily driving Ghostty).

The pr adds an action called `toggle_visibility`, which essentially hides and shows all Ghostty windows with a global key (and from the menubar).

Commit:
Add action on Zig side

Commit:
Add hiding toggle, hook up to menu / shortcut

Commit:
Fix typo

mitchellh reviewed (changes requested):
Looks great! Some minor requests...

Commit:
Refactor to hide only BaseTerminalController windows

This also slightly changes the code, as the duplication of the for loop
was making it harder to read now. I think technically slightly less
efficient, but this is hardly a hot code path, so should be fine imo.

Commit:
Review Feedback - formatting / docs

mitchellh committed:
docs updates
